### ISSUE TITLE
[Bug] Failing initial render of a focused immediate child component style

### ISSUE THREAD

**Issue by davep** [Textualize/textual#1852]
Little bit of a niche one this one, but likely worth tracking down. Consider the following (slightly contrived, but for a reason) code:

```python
from rich.text import Text

from textual.app        import App, ComposeResult, RenderResult
from textual.containers import Vertical
from textual.widgets    import Header, Footer
from textual.widget     import Widget

class Tester( Widget, can_focus=True ):

    COMPONENT_CLASSES = {
        "tester--text"
    }

    DEFAULT_CSS = """
    Tester {
        height: auto;
    }

    Tester:focus > .tester--text {
        background: red;
    }
    """

    def render( self ) -> RenderResult:
        return Text(
            "This is a test widget",
            style=self.get_component_rich_style( "tester--text" )
        )

class StyleBugApp( App[ None ] ):

    def compose( self ) -> ComposeResult:
        yield Header()
        yield Vertical( *[ Tester() for _ in range( 40 ) ] )
        yield Footer()

if __name__ == "__main__":
    StyleBugApp().run()
```

It should create a widget that contains a collection of lines, each of which will have a red background when focus moves into them. The key here is that the `Tester` widget doesn't style itself, but styles a renderable with a component style.

The issue here is that when you run up the application, then press <kbd>TAB</kbd> once, the first item doesn't style as having focus. However, the moment you cause a refresh -- by resizing the terminal for example -- it does correctly highlight the first item.

If I add something like this to the CSS:

```scss
    Tester:focus {
        display: block;
    }
```

it works fine every time.

In short:

- `:focus` rule on an immediate child component class of a widget alone: problematic
- `:focus` rule on on the widget alone: things work
